CMakeLists.txt

```diff
@@ -123,10 +123,55 @@ endif()
 # Generate distribution packages with CPack.
 if(WIN32)
     set(CPACK_GENERATOR ZIP)
+elseif(LINUX)
+    set(CPACK_GENERATOR External)
+	set(CPACK_EXTERNAL_ENABLE_STAGING YES)
+	set(CPACK_EXTERNAL_PACKAGE_SCRIPT "${PROJECT_BINARY_DIR}/appimage-generate.cmake")
+
+	file(GENERATE
+	OUTPUT "${PROJECT_BINARY_DIR}/appimage-generate.cmake"
+	CONTENT [[
+	include(CMakePrintHelpers)
+	cmake_print_variables(CPACK_TEMPORARY_DIRECTORY)
+	cmake_print_variables(CPACK_TOPLEVEL_DIRECTORY)
+	cmake_print_variables(CPACK_PACKAGE_DIRECTORY)
+	cmake_print_variables(CPACK_PACKAGE_FILE_NAME)
+
+	find_program(LINUXDEPLOY_EXECUTABLE
```

Comment:
What is this script and why do we have to execute it?

Reply:
`Linuxdeploy` is the "external" packager. It creates the appimage packaging format.

It makes more sense to use this for Release. I added it in the CI, since the release was specifically for Windows Artifacts `win_msvc.yml`, but this could be changed around.

Reply:
Yes, please add this to the release. Feel free to rename that file.

Reply:
Well, this is going to be messy, `win_msvc.yml` contains a lot of Windows specific stuff. I suggest merging this as is, I'll add the release to `main.yml` and other changes.

Reply:
> Well, this is going to be messy, `win_msvc.yml` contains a lot of Windows specific stuff. I suggest merging this as is, I'll add the release to `main.yml` and other changes.

Oh, agreed. Thank you!